### PR TITLE
[6.11.z] Port/refactor of f-m health tests from testfm

### PR DESCRIPTION
**Pull request by Satellite-QE**
Cherrypick of PR: https://github.com/SatelliteQE/robottelo/pull/10288

Description:

1. Port/refactor f-m health subcommand tests from TestFM to robottelo
https://github.com/SatelliteQE/testfm/blob/master/tests/test_health.py

It includes changes from https://github.com/SatelliteQE/robottelo/pull/10227, and it will be rebased when either PR is merged

### CODE REVIEW

File: tests/foreman/maintain/test_health.py [hunk 1]
```diff
@@ -0,0 +1,827 @@
...
+        1. configure yum exclude.
+        2. Run satellite-maintain health check --label validate-yum-config
+        3. Assert that excluded packages are listed in output.
+        4. remove yum exclude configured in step 1.
+
+    :expectedresults: validate-yum-config should work.
+
+    :BZ: 1669498
+
+    :customercenario: true
```
Comment (Gauravtalreja1) [inline tests/foreman/maintain/test_health.py:337]: typo `s/customercenario/customerscenario`

File: tests/foreman/maintain/test_health.py [hunk 2]
```diff
@@ -0,0 +1,827 @@
...
+    :expectedresults: check-hotfix-installed check should detect modified file
+        and installed hotfix.
+    """
+    # Verify check-hotfix-installed without hotfix package.
+    result = sat_maintain.cli.Health.check(options={'label': 'check-hotfix-installed'})
+    assert result.status == 0
+    assert 'WARNING' not in result.stdout
+
+    # Verify check-hotfix-installed with hotfix package.
+    gems_path = '/usr/share/gems/gems/'
```
Comment (Gauravtalreja1) [inline tests/foreman/maintain/test_health.py:299]: This path will change for el7 and el8
```suggestion
    gems_path = f"""{f"{'/opt/theforeman/tfm/root' if sat_maintain.os_version.major < 8 else ''}" + '/usr/share/gems/gems/'}"""
```

**Review by Gauravtalreja1 — CHANGES_REQUESTED** [review]

**Comment by Gauravtalreja1** [comment]
@shubhamsg199 Could you please update this PR, and run PRT?

File: tests/foreman/maintain/test_health.py (hunk 2 above)
Comment (shubhamsg199) [reply tests/foreman/maintain/test_health.py]: Updated.

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/maintain/test_health.py

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/maintain/test_health.py
deploy_rhel_version: 7

**Comment by Gauravtalreja1** [comment]
trigger: test-robottelo
pytest: tests/foreman/maintain/test_health.py
deploy_rhel_version: 7

**Comment by shubhamsg199** [comment]
There were 4 test failures 2 of them related epel install which are fixed with recent commit remaining two are intermittent failures.

**Comment by shubhamsg199** [comment]
**Test Result for failed epel test:**
```
pytest -k test_positive_health_check_epel_repository tests/foreman/maintain/test_health.py
============================================= test session starts ==============================================================
platform linux -- Python 3.8.6, pytest-7.1.3, pluggy-0.13.1
shared_function enabled - OFF - scope:  - storage: file
rootdir: /home/sganar/satellite/robottelo, configfile: pyproject.toml
plugins: services-2.2.1, forked-1.3.0, cov-3.0.0, xdist-2.5.0, ibutsu-2.2.4, fixture-tools-1.1.0, reportportal-5.1.2, lazy-fixture-0.6.3, mock-3.10.0
collected 33 items / 32 deselected / 1 selected                                                                                                                                                                                       

tests/foreman/maintain/test_health.py .                                                                                                                                                                                               [100%]

====================================== 1 passed in 68.83s (0:01:08) ============================================
```

**Review by Gauravtalreja1 — APPROVED** [review]
Nice job on handling changes for el7 and el8

**Pull request merged**